bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/client/ParameterObjectArgumentResolver.kt

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.gate.api.client
+
+import org.springdoc.core.annotations.ParameterObject
+import org.springframework.core.MethodParameter
+import org.springframework.format.support.DefaultFormattingConversionService
+import org.springframework.web.service.invoker.HttpRequestValues
+import org.springframework.web.service.invoker.HttpServiceArgumentResolver
+import kotlin.reflect.full.memberProperties
+
+class ParameterObjectArgumentResolver : HttpServiceArgumentResolver {
```

Comment:
Needs to be merged with the one in bpdm-pool-api and extracted to the bpdm-common module